test/unit/L2/delegatorPool.test.ts

```diff
@@ -0,0 +1,246 @@
+import {FakeContract, smock} from '@defi-wonderland/smock';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/dist/src/signers';
+import {expect, use} from 'chai';
+
+import {ethers} from 'hardhat';
+import {DelegatorPool, DelegatorPool__factory} from '../../../typechain';
+
+use(smock.matchers);
+
+describe('DelegatorPool', function() {
+  let delegatorPool: DelegatorPool;
+
+  let bondingManagerMock: FakeContract;
+
+  let delegator: SignerWithAddress;
+  let delegator1: SignerWithAddress;
+  let delegator2: SignerWithAddress;
+  let mockL2MigratorEOA: SignerWithAddress;
+  let mockBondingManagerEOA: SignerWithAddress;
+
+  class StakeAndFees {
+    public claimedStake = 0;
+    public pendingStake;
+    public pendingFees;
+
+    public sequence = 0;
+
+    constructor(
+      public readonly initialStake: number,
+      public readonly initialFees: number,
+    ) {
+      this.pendingStake = initialStake;
+      this.pendingFees = initialFees;
+    }
+
+    calculateClaim(stake: number) {
+      const owedStake =
+        (this.pendingStake * stake) / (this.initialStake - this.claimedStake);
+
+      const owedFees =
+        (this.pendingFees * stake) / (this.initialStake - this.claimedStake);
+
+      return {
+        owedStake,
+        owedFees,
+      };
+    }
+
+    addRewards(stake: number, fees: number) {
```

Comment:
Would a more descriptive name for this function be `updateStakeAndFees()`?

Reply:
fixed in 7be5149